discretisedfield/tools/tools.py

```diff
@@ -290,18 +299,24 @@ def topological_charge(field, /, method="continuous", absolute=False):
             return float(q.integrate())
 
     elif method == "berg-luescher":
-        axis1 = field.mesh.attributes["axis1"]
-        axis2 = field.mesh.attributes["axis2"]
+        # Spatial axes associated with the first and second vector
+        # dimensions.
+        axis1 = field.vdim_mapping[field.vdims[0]]
+        axis2 = field.vdim_mapping[field.vdims[1]]
```

Comment:
Again, `axis1` and `axis2` are not guaranteed to be valid and we should maybe add an explicit check.